src/beanmachine/ppl/inference/monte_carlo_samples.py

```diff
@@ -273,8 +273,15 @@ def to_inference_data(self, include_adapt_steps: bool = False) -> az.InferenceDa
 
         if "posterior" in self.namespaces:
             posterior = self.namespaces["posterior"].samples
+            posterior = {
+                key: value.detach().numpy() for key, value in posterior.items()
+            }
```

Comment:
Can we refactor this into a utility function so we don't need to repeat the same set of code for each field?

Reply:
Created a new utility method, and created a new test for it.